cmake/wasm-app.cmake

```diff
@@ -0,0 +1,58 @@
+include(cmake/common.cmake)
+
+# lib use cmake?
+# add_subdirectory(lib)
+set(EXAMPLE_LINK_OPTIONS
+    ${COMMON_FLAGS}
+    -sLLD_REPORT_UNDEFINED
+    -sALLOW_MEMORY_GROWTH=1
+    -sFORCE_FILESYSTEM=0
+    -sFILESYSTEM=0
+    -sNO_EXIT_RUNTIME=1
+    -sINITIAL_MEMORY=128MB
+    -sWASM=1
+    -sUSE_WEBGL2=1
+    -sDEMANGLE_SUPPORT=1
+    -sASSERTIONS=1
+    -sGL_ASSERTIONS=1
+    -sUSE_SDL=2
+)
+
+set(EXAMPLE_INCLUDE_DIRECTORIES
+    .
+    ${PROJECT_BINARY_DIR}
+    ${COMMON_INCLUDE_DIRECTORIES}
+)
+
+add_executable(wasm-example
+    ${SRCS}
+)
+
+target_link_libraries(wasm-example PUBLIC
+)
+
+# add_definitions()
+add_custom_command(TARGET wasm-example POST_BUILD
+
+    # 好像是有什么bug，导致后缀是js，而不是html，先直接改名？
```

Comment:
So you mean the `--oformat=html` below doesn't do anything?

May I ask what contains the output artifact name? (in other words, how is `-o main.html` achieved?)

Reply:
for the output named xxx.html,
1st, The ‘xxx’ is given by the command of cmake named add_executable
2nd, The actual format is given by the option of '--oformat'
so, --oformat=html below  do write things,
But, The only bug is the suffix of the file, which should has suffix html instead of .js